marshal/consumer.go

```diff
@@ -316,11 +325,14 @@ func (c *Consumer) rndIntn(n int) int {
 // set on aggressive, this will try to claim ALL partitions that are free. Balanced mode
 // will claim a single partition.
 func (c *Consumer) claimPartitions() {
+	c.lock.RLock()
```

Comment:
For this we actually prefer to run things in a closure and use deferred unlocks. I.e.,

```
func() {
    c.lock.RLock()
    defer c.lock.RUnlock()

    do stuff
}()
```